Sources/core/objects/Projectiles.cpp

```diff
@@ -0,0 +1,254 @@
+#include "Projectiles.h"
+#include "../Helpers.h"
+
+
+Projectile::Projectile(size_t id) : IMovable(id)
+{
+    SetPassable(D_ALL, Passable::AIR);
+    tickSpeed = 0.25;
+    damage = 0;
+    SetSprite("icons/projectiles.dmi");
+    v_level = 5;
+}
+void Projectile::Process()
+{
+    if(set_target_ == 0)
+    {
+       return;
+    }
+    int z = current_step_;
+    Dir step = VDirToDir(movement_[z]);
+
+   if(!TryMove(step))
+   {
+       this->Delete();
+       return;
+   }
+   current_step_++;
+   if(!(current_step_ < movement_.size()) && !reached_target )
+   {
+       current_step_ = 0;
+       VDir temp = movement_[0];
+       movement_.clear();
+       reached_target = true;
+       movement_.push_back(temp);
+   }
+   if(!(current_step_<movement_.size()) && reached_target )
+   {
+   current_step_ = 0;
+   }   
+}
+
+bool Projectile::CheckPassable()
+{
+    id_ptr_on<Projectile> p = this->GetId();
```

Comment:
```
Human::Bump(id_ptr_on<IMovable> object)
{
    if (id_ptr_on<Projectile> projectile = object)
    {
        health -= projectile->damage_;
        return;
    }
    IMob::Bump(object);
}
```
That is how it should work. You may add new variable in `Projectile` which will correspond the amount of damage. For a while there is only one type of damage. It will be changed in the future, but it is a separate task.